GrayBMP.cs

```diff
@@ -74,6 +74,29 @@ class GrayBMP {
       Dirty (x1, y1); Dirty (x2, y2);
    }
 
+   /// <summary>Draw a series of horizontal line given a y cooridnate using ordered x values</summary>
+   public void DrawHorizontalLines (int y, int gray, params int[] xValues) {

```

Comment:
This is inefficient - each call to this will require an allocation (because of the params int[]). It is better to have a simpler signature like `DrawHorizontalLine (int x1, int y1, int y, int gray)` and call it repeatedly.